tests/phpunit/tests/image/editorImagick.php

```diff
@@ -641,4 +641,43 @@ public function test_directory_creation() {
 
 		$this->assertNotWPError( $saved );
 	}
+
+	/**
+	 * @ticket 39216
+	 * Only affects systems with Ghostscript version >= 9.14.
+	 */
+	public function test_remove_alpha_pdf_preview() {
+		if ( ! wp_image_editor_supports( array( 'mime_type' => 'application/pdf' ) ) ) {
+			$this->markTestSkipped( 'Rendering PDFs is not supported on this system.' );
+		}
+
+		$test_file     = DIR_TESTDATA . '/images/test-alpha.pdf';
+		$attachment_id = $this->factory->attachment->create_upload_object( $test_file );
+		$this->assertNotEmpty( $attachment_id );
+
+		$attached_file = get_attached_file( $attachment_id );
+		$this->assertNotEmpty( $attached_file );
+
+		$rgb = array(
+			'r' => true,
+			'g' => true,
+			'b' => true,
+		);
+
+		// White.
+		$expected = array(
+			'r' => 1,
+			'g' => 1,
+			'b' => 1,
+		);
+
+		$check = image_get_intermediate_size( $attachment_id, 'full' );
+		$this->assertNotEmpty( $check['file'] );
+
+		$check_file = path_join( dirname( $attached_file ), $check['file'] );
+		$imagick    = new Imagick( $check_file );
+		$output     = array_map( 'round', array_intersect_key( $imagick->getImagePixelColor( 100, 100 )->getColor( true /*normalized*/ ), $rgb ) );
+		$imagick->destroy();
+		$this->assertEquals( $expected, $output ); // Allow for floating point equivalence.
```

Comment:
As we're trying to reduce the use of `assertEquals()` in the test suite unless it's really needed (such as when comparing objects / object arrays), is there a way we can enforce a specific type so that we can replace `assertEquals()` with `assertSame()` here?

Maybe this?

```php
$output = array_map(
	static function( $value ) {
		return (int) round( $value );
	},
	array_intersect_key( $imagick->getImagePixelColor( 100, 100 )->getColor( true /* normalized */ ), $rgb )
);
```